File: GolangSolutions/Array/M2073_买票需要的时间.go
/*
	@author Hyperspace
	@date 2024/09/29
	@file M2073_买票需要的时间.go
*/

package Array

func timeRequiredToBuy(tickets []int, k int) (ans int) {
	s := tickets[k]
	for i, t := range tickets {
		if i > k {
			ans += min(t, s-1)
		} else {
			ans += min(t, s)
		}
	}

	return
}

File: CPPSolutions/LeetCode/Array/M2073_买票需要的时间.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * @author Hyper
 * @date 2024/09/29
 * @file M2073_买票需要的时间.cpp
 */
class Solution {
public:
    int timeRequiredToBuy(vector<int> &tickets, int k) {
        int n = tickets.size();
        int ans = 0;
        for (int i = 0; i < n; i++) ans += min(tickets[i], tickets[k] - (i > k));

        return ans;
    }
};